pkg/gameservers/health.go

```diff
@@ -126,7 +145,23 @@ func (hc *HealthController) syncGameServer(key string) error {
 		return errors.Wrapf(err, "error retrieving GameServer %s from namespace %s", name, namespace)
 	}
 
-	if gs.Status.State == v1alpha1.Ready {
+	var reason string
+	unhealthy := false
+
+	switch gs.Status.State {
+
+	case v1alpha1.Starting:
+		hc.logger.WithField("key", key).Info("GameServer cannot start on this port")
+		unhealthy = true
+		reason = "No more of the allocated ports free"
```

Comment:
Your method and test works with `free ports` but your real code return a string with `ports free`. These need to be aligned.

Reply:
Done!